src/ODESolvers/MultirateRungeKuttaMethod.jl

```diff
@@ -0,0 +1,138 @@
+module MultirateRungeKuttaMethod
+using ..ODESolvers
+ODEs = ODESolvers
+using ..LowStorageRungeKuttaMethod
+using ..MPIStateArrays: device, realview
+
+using GPUifyLoops
+include("MultirateRungeKuttaMethod_kernels.jl")
+
+export MultirateRungeKutta
+
+"""
+    MultirateRungeKutta(slow_RK_solver, fas_RKt_solver; dt, t0 = 0)
+
+This is a time stepping object for explicitly time stepping the differential
+equation given by the right-hand-side function `f` with the state `Q`, i.e.,
+
+```math
+  \\dot{Q} = f_fast(Q, t) + f_slow(Q, t)
+```
+
+with the required time step size `dt` and optional initial time `t0`.  This
+time stepping object is intended to be passed to the `solve!` command.
+
+The constructor builds a multirate Runge-Kutta scheme using two different RK
+solvers. This is based on 
+
+Currently only the following low storage RK methods can be used as fast and slow
+solvers
+
+  - [`LSRK54CarpenterKennedy`](@ref)
+  - [`LSRK144NiegemannDiehlBusch`](@ref)
+
+### References
+
+    @article{SchlegelKnothArnoldWolke2012,
+      title={Implementation of multirate time integration methods for air
+             pollution modelling},
+      author={Schlegel, M and Knoth, O and Arnold, M and Wolke, R},
+      journal={Geoscientific Model Development},
+      volume={5},
+      number={6},
+      pages={1395--1405},
+      year={2012},
+      publisher={Copernicus GmbH}
+    }
+"""
+struct MultirateRungeKutta{SS, FS, RT} <: ODEs.AbstractODESolver
+  "fast solver"
+  slow_solver::SS
+  "slow solver"
+  fast_solver::FS
```

Comment:
Comment mismatch "fast solver" before `slow_solver` and "slow_solver" before `fast_solver`

Reply:
fixes